src/css/finalPropName.js

```diff
@@ -0,0 +1,39 @@
+define( [ "../var/document" ], function( document ) {
+
+"use strict";
+
+var cssPrefixes = [ "Webkit", "Moz", "ms" ],
+	emptyStyle = document.createElement( "div" ).style,
+	vendorProps = {};
+
+// Return a vendor-prefixed property or undefined
+function vendorPropName( name ) {
+
+	// Check for vendor prefixed names
+	var capName = name[ 0 ].toUpperCase() + name.slice( 1 ),
+		i = cssPrefixes.length;
+
+	while ( i-- ) {
+		name = cssPrefixes[ i ] + capName;
+		if ( name in emptyStyle ) {
+			return name;
+		}
+	}
+}
+
+// Return a potentially-mapped jQuery.cssProps or vendor prefixed property
+function finalPropName( name ) {
+	var final = jQuery.cssProps[ name ] || vendorProps[ name ];
```

Comment:
Hm.. I don't think this resolved the concern from https://github.com/jquery/jquery-migrate/pull/291.

The jquery-migrate pull proposes a warning for getting (and theoretically, setting) of `jQuery.cssProps` itself. That may seem odd, but is actually fine, given that detecting the setting of sub-properties is hard, and, in order to set a sub-property, one has to get the cssProps object first, which will trigger the warning. All good.

However, the code here in core/css.js, still gets the `jQuery.cssProps` object the same way as before, so it seems that would trigger the warning?

The only thing that changed is it now gets the object once instead of twice.

Reply:
In the previous code, we **set** a property on `jQuery.cssProps`. Now we no longer do that, so Migrate can look for situations where external code sets values here.

Reply:
Yep, but the warning added in https://github.com/jquery/jquery-migrate/pull/291 is not for getting/setting unknown properties on `jQuery.cssProps`, it warns for getting (and setting) of the `cssProps` property of `jQuery`.